src/core/Akka.TestKit/Internal/ChannelQueue.cs

```diff
@@ -20,33 +22,41 @@ namespace Akka.TestKit.Internal
     /// <remarks>Note! Part of internal API. Breaking changes may occur without notice. Use at own risk.</remarks>
     /// </summary>
     /// <typeparam name="T">The type of item to store.</typeparam>
-    public class BlockingQueue<T>
+    public class ChannelQueue<T>
     {
-        private readonly BlockingCollection<Positioned> _collection = new BlockingCollection<Positioned>(new QueueWithAddFirst());
+        private readonly Channel<Positioned> _channel = Channel.CreateBounded<Positioned>(2000);
+
 
         /// <summary>
         /// The number of items that are currently in the queue.
         /// </summary>
-        public int Count { get { return _collection.Count; } }
+        public int Count { get { return _channel.Reader.Count; } }
 
         /// <summary>
         /// Adds the specified item to the end of the queue.
         /// </summary>
         /// <param name="item">The item to add to the queue.</param>
-        public void Enqueue(T item)
+        public async ValueTask Enqueue(T item)
```

Comment:
Incomplete implementation, should create a new method that accept a CancellationToken and figure out a way to guard this method with a time out mechanism